### ISSUE TITLE
Include keyframes in mixins (scss)

### ISSUE THREAD

**Issue by FlTr** [IanLunn/Hover#61]
Think it would be great if the keyframes directives are included in the mixins. 

```scss
@mixin pulse {
  @at-root {
    @include keyframes( #{$nameSpace}-pulse ) {
      // ...
    }
  }

 @include hacks();

  // ...
}
```

For individual usage a _hover.scss partial would be nice to prevent Copy&Paste actions.

**Comment by artiswilliams** [comment]
I agree.  I don't like seeing all that keyframe code in my css when I'm not using the effect.

If the keyframe directive is in the mixin and $includeClasses is set to false then no css is generated unless a mixin is actually used somewhere.

**Comment by IanLunn** [comment]
Agreed. I'll add this to the next release. Any objection to this being done without the use of `@at-root` now it's not necessary in Sass 3.4+

Does LESS have a similar directive?

**Comment by FlTr** [comment]
Not sure if it works right without ```@at-root```. 

Sorry, no experience with LESS

**Comment by qzminski** [comment]
Hey @IanLunn any news on this? It would be a really great improvement.

**Comment by IanLunn** [comment]
@qzminski I plan to add this as part of 3.0 but no date for this.

**Comment by Hussseinkizz** [comment]
The docs for how to use in sass for a beginner like me are totally not well explained, like how do I actually use the effect?
